tutorials/en/vectara-advanced-app-tutorial-showcase-the-creation-of-vectara-app-in-legal-or-customer-support-use-case.mdx

```diff
@@ -0,0 +1,671 @@
+
+#### Vectara’s Indexing and Querying APIs: The Navigators
+
+When an end-user sends a query, Vectara’s state-of-the-art indexing and querying APIs spring into action. They ingest the data, embed it, and whisk through the corpus to fetch the most apropos response. This data is then fed to a summarizer, adding a human touch to the output, thus evading the robotic undertone often associated with bot responses.
+
+#### Implementation with Streamlit: The Playground
+
+To breathe life into our concept, we’ll employ Streamlit which not only unveils the inner workings of the code but provides a playground to test and iterate quickly. As we advance, a treasure trove of Vectara libraries awaits the backend developers, promising a smoother sail even if centering that div seems like chasing the horizon!
+
+#### A Dash of Humor: The Artistic Struggle
+
+Oh, and about the artistic struggle with centering divs, fear not! While art may have its Mona Lisa, in the coding world, a perfectly centered div is no less of a masterpiece! (And just like me, it seems Vectara too, isn’t too fond of going off-center!)
+
+---
+
+### Setting the Stage: Setup and Installation Guide 🛠️
+
+Before we delve into the realms of code and explore the intricacies of our application, it's imperative to set the stage right. This segment is dedicated to guiding you through the process of setting up and installing the necessary components for our application. The emphasis is on ensuring a smooth sail as we venture into the development phase.
+
+#### Step 1: Create a Virtual Environment
+Creating a virtual environment is a good practice to manage dependencies and ensure the application runs consistently across different setups.
+
+```bash
+python3 -m venv vectara-env
+```
+
+Activate the virtual environment:
+- On Windows:
+```bash
+.\vectara-env\Scripts\activate
+```
+- On macOS and Linux:
+```bash
+source vectara-env/bin/activate
+```
+
+#### Step 2: Install Necessary Packages
+Install the necessary packages using pip:
+
+```bash
+pip install streamlit python-dotenv requests authlib
+```
+
+#### Step 3: Create the .env File
+Create a file named `.env` in the root directory of your project. This file will store your environment variables. Here's how your `.env` file should look like, replace the placeholders with your actual credentials:
+
+```plaintext
+AUTH_URL=https://your-auth-url.com
+APP_CLIENT_ID=your-app-client-id
+APP_CLIENT_SECRET=your-app-client-secret
+CUSTOMER_ID=your-customer-id
+CORPUS_ID=your-corpus-id
+IDX_ADDRESS=api.vectara.io
+API_KEY=your-api-key
+```
+
+Make sure you copy/add the IDX_Address
+
+---
+
+#### Step 4: Setup Instructions
```

Comment:
It may be helpful to explain in the below why we use both API-key and OAuth. As I understand it, in this example, OAuth is used for indexing and API-key for searching. But please let's add some more text to explain that choice and also make sure folks understand you can also index with API-key as another option.